master/buildbot/test/unit/test_schedulers_triggerable.py

```diff
@@ -101,12 +102,12 @@ def assertTriggeredBuildset(self, idsDeferred, waited_for, properties={}, source
         for expected_ss, actual_ss in zip(sourcestamps, actual_sourcestamps):
             actual_ss = actual_ss.copy()
             # We don't care if the actual sourcestamp has *more* attributes than expected.
-            for key in actual_ss.keys():
+            for key in list(actual_ss):
```

Comment:
I think that should actually be ```list(actual_ss.keys())```, since we need a copy of the keys. We can't use the actual dictionary object since it is modified in the loop and it would raise an exception